### PR TITLE
updated styles to follow BEM notation #24 

### PR DESCRIPTION
**Pull request by diliger**
#24 
#78 
#79

### CODE REVIEW

**Comment by aloscha** [comment]
Sometimes image is really too small:
image

The image preview is also too small
image

In previous ImageApp version:
image

**Comment by diliger** [comment]
changed min height for images.

File: src/Images/wwwroot/Images/css/images.css [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    border-radius: 3px;
+    text-align:center;
+}
+
+.images-content-preview__image__thumbnail {
+    max-height: 500px;
+    max-width: 100%;
+}
+
+.images-content-preview__video__preview {
```
Comment (warpech) [inline src/Images/wwwroot/Images/css/images.css]: `images-content-preview__image__thumbnail` and `images-content-preview__video__preview` are not correct BEM names (there can be only one "Element" in a class name)

Also, I think that "thumbnail" is not a correct term in this view. We only have thumbnails in a carousel.

I propose the following names for these classes: `images-content-preview__image-content` and `images-content-preview__video-content`.

File: src/Images/wwwroot/Images/css/standalone.css [hunk 2]
```diff
@@ -1,87 +0,0 @@
...
-    top: 5px;
-    position: absolute
-}
-
-.images-illustration-simple-thumbnail {
-    background-size: contain;
-    background-position: left top;
-    background-repeat: no-repeat;
-    flex: 1 1 auto;
-}
```
Comment (warpech) [inline src/Images/wwwroot/Images/css/standalone.css]: All of these classes were not used?

File: src/Images/wwwroot/Images/viewmodels/ConceptPage.html [hunk 3]
```diff
@@ -1,14 +0,0 @@
...
-        <so-juicy-filedrop session-id="{{model.SessionId}}" upload-url="/images/images" value="{{model.ImageURL$}}" max-file-size="{{model.MaxFileSize}}" allowed-mime-types="{{model.AllowedMimeTypes}}">
-            <template is="dom-if" if="{{model.ImageURL$}}">
-                <div>
-                    <button class="btn btn-default btn-xs" onclick="++this.value;" value="{{model.Delete$::click}}">Clear</button>
-                    <button class="btn btn-primary btn-xs" onclick="++this.value;" value="{{model.Save$::click}}">Save</button>
-                </div>
-            </template>
-        </so-juicy-filedrop>
-    </template>
-</template>
```
Comment (warpech) [inline src/Images/wwwroot/Images/viewmodels/ConceptPage.html]: So, this file was totally unused? Great find!

File: src/Images/wwwroot/Images/viewmodels/ContentPage.html [hunk 4]
```diff
@@ -57,17 +39,17 @@
             var template = script.previousElementSibling;
 
             template.getImageStyle = function (url) {
-                return "background-image:url('" + url + "')";
+                return url;
             };
             template.getFileStyle = function () {
-                return "background-image:url('/images/css/file_preview.png')";
+                return "/images/css/file_preview.png";
```
Comment (warpech) [inline src/Images/wwwroot/Images/viewmodels/ContentPage.html]: How is this supposed to work? It will result in `style="/images/css/file_preview.png"`

File: src/Images/wwwroot/Images/viewmodels/ContentPage.html [hunk 5]
```diff
@@ -36,16 +17,17 @@
             </template>
             <template is="dom-if" if="{{!model.IsVideo}}" restamp="true">
                 <template is="dom-if" if="{{model.IsImage}}">
-                    <div class="img-border content-image" on-mousedown="imageClickEvent">
-                        <div class="img-thumbnail" style$="{{getImageStyle(model.URL)}}"></div>
+                    <div class="images-content-preview__image" on-mousedown="imageClickEvent">
+                        <!--<div class="images-content-preview__image__thumbnail" style$="{{getImageStyle(model.URL)}}"></div>-->
+                        <img src$="{{getImageStyle(model.URL)}}" alt="{{model.Name}}" class="images-content-preview__image__thumbnail" />
```
Comment (warpech) [inline src/Images/wwwroot/Images/viewmodels/ContentPage.html]: Such change was not expected in BEM-ification. Is this fixing some problem? `<div class="img-thumbnail" style$="{{getImageStyle(model.URL)}}"></div>` was working differently (`background-size: contain`) and nobody reported problems with it.

File: src/Images/wwwroot/Images/viewmodels/ContentPage.html [hunk 6]
```diff
@@ -93,11 +75,11 @@
     </script>
     <template is="starcounter-composition">
         <style>
-            .content-preview {
-                height: 100%;
+            .images-content-preview {
+                
             }
         </style>
```
Comment (warpech) [inline src/Images/wwwroot/Images/viewmodels/ContentPage.html]: Looks like this class can be removed.

File: src/Images/wwwroot/Images/viewmodels/SettingsPage.html [hunk 7]
```diff
@@ -1,18 +1,30 @@
...
+        font-size: 12pt;
+        margin: 0;
+    }
+    .images-settings__control {
+        width: 50%
+    }
+    .images-settings__button {
+        margin-top: 20px;
+    }
+</style>
```
Comment (warpech) [inline src/Images/wwwroot/Images/viewmodels/SettingsPage.html]: The above classes are not correct BEM. Suggested names: `images-settings-label`, `images-settings-control`, `images-settings-button`

File: src/Images/wwwroot/Images/css/standalone.css (hunk 2 above)
Comment (diliger) [inline src/Images/wwwroot/Images/css/standalone.css]: Yes, standalone.css wasn't used.

File: src/Images/wwwroot/Images/viewmodels/ConceptPage.html (hunk 3 above)
Comment (diliger) [inline src/Images/wwwroot/Images/viewmodels/ConceptPage.html]: Yes, it was excluded from project some time ago.

File: src/Images/wwwroot/Images/viewmodels/ContentPage.html (hunk 5 above)
Comment (diliger) [inline src/Images/wwwroot/Images/viewmodels/ContentPage.html]: Yes, @aloscha reported a problem about too small images.
I tried to find a good way with `background-images` but finally ended up with ``

File: src/Images/wwwroot/Images/viewmodels/ContentPage.html (hunk 4 above)
Comment (aloscha) [inline src/Images/wwwroot/Images/viewmodels/ContentPage.html]: Based on code, it should be removed anyway.

Comment (diliger) [inline src/Images/wwwroot/Images/viewmodels/ContentPage.html]: Fixed.
`file_preview.png` is used to make a preview for the content that can't be previewed with image or video for whatever reason.

File: src/Images/wwwroot/Images/viewmodels/ContentPage.html (hunk 6 above)
Comment (diliger) [inline src/Images/wwwroot/Images/viewmodels/ContentPage.html]: Removed.

File: src/Images/wwwroot/Images/viewmodels/SettingsPage.html (hunk 7 above)
Comment (diliger) [inline src/Images/wwwroot/Images/viewmodels/SettingsPage.html]: Renamed.

File: src/Images/wwwroot/Images/viewmodels/ContentPage.html (hunk 5 above)
Comment (warpech) [inline src/Images/wwwroot/Images/viewmodels/ContentPage.html]: I don't understand. This PR should only refactor classes and stylesheets. How come it also fixes a problem with too small images? That should be a separate issue and a separate PR.

File: src/Images/wwwroot/Images/viewmodels/ContentPage.html [hunk 8]
```diff
@@ -36,16 +17,16 @@
...
                     <paper-dialog modal="true" opened="[[model.IsLightBoxVisible$]]" with-backdrop="true" class="images-preview-image-dialog" on-mousedown="closeEvent">
-                        <img src="{{model.URL}}" class="images-preview-image-dialog-image"/>
+                        <img src="{{model.URL}}" class="images-preview-image-dialog-image" />
                     </paper-dialog>
                 </template>
                 <template is="dom-if" if="{{!model.IsImage}}">
-                    <div class="img-border content-image">
-                        <div class="img-thumbnail" style$="{{getFileStyle()}}"></div>
+                    <div class="images-content-preview__image">
+                        <div class="images-content-preview__image-content" style$="{{getFileStyle()}}"></div>
```
Comment (warpech) [inline src/Images/wwwroot/Images/viewmodels/ContentPage.html]: Honestly, this can be changed to:

```
<div class="images-content-preview__image-content"></div>
```

With `template.getFileStyle` removed and `background-image:url('/images/css/file_preview.png')` added to the CSS file.

File: src/Images/wwwroot/Images/viewmodels/ContentPage.html (hunk 5 above)
Comment (diliger) [inline src/Images/wwwroot/Images/viewmodels/ContentPage.html]: Yes, my mistake.

File: src/Images/wwwroot/Images/viewmodels/ContentPage.html (hunk 8 above)
Comment (diliger) [inline src/Images/wwwroot/Images/viewmodels/ContentPage.html]: Changed.

**Comment by aloscha** [comment]
The size is ok for me
image

**Comment by diliger** [comment]
@aloscha you mean after my fix?

**Comment by warpech** [comment]
ping @aloscha 

@diliger could you pls merge current `develop` to `BEM-styles` branch? There are some conflicts after https://github.com/StarcounterApps/Images/pull/88

**Comment by aloscha** [comment]
>@aloscha you mean after my fix?
yes

**Comment by diliger** [comment]
Resolved conflicts.

**Comment by aloscha** [comment]
@warpech could you check and if everything will be ok I'll merge it?

**Comment by warpech** [comment]
I did not get an answer for the question https://github.com/StarcounterApps/Images/pull/81#discussion_r109651182, that this PR fixes some image rendering problem that was actually not reported as an issue.

@un-tone could you pls review if the changes in https://github.com/StarcounterApps/Images/blob/0005b97cd9eceb2d64762d5b2ba24a8dab5d219e/src/Images/wwwroot/Images/viewmodels/ContentPage.html don't break the use cases known to you?

The BEM part looks ok to me.

**Comment by un-tone** [comment]
@warpech LGTM

**Comment by warpech** [comment]
@diliger then it lgtm too. Pls merge
